### PR TITLE
libnixf: line column range

### PR DESCRIPTION
**Pull request by inclyc**
- [x] fixup test cases

### CODE REVIEW

File: libnixf/include/nixf/Basic/Range.h [hunk 1]
```diff
@@ -5,16 +5,29 @@
...
 namespace nixf {
 
-struct OffsetRange {
-  const char *Begin;
-  const char *End;
+struct Point {
+  int64_t Line;
+  int64_t Column;
+  std::size_t Offset;
+};
```
Comment (Origami404) [inline libnixf/include/nixf/Basic/Range.h:12]: I believe that we should define the meanings of these fields more carefully in the comments here, such as:

- Clearly specify whether a line is terminated by `\r`, `\n`, or `\r\n`. (This also defines where is the begin of a line)
- Clearly define whether 'Column' refers to the character offset from the beginning of the line or the byte offset.
- Clearly define whether 'Offset' refers to the character offset or the byte offset.
- How can we ensure that Line/Column matches the Offset?

Comment (Origami404) [reply libnixf/include/nixf/Basic/Range.h:12]: Or just add a comment: `// See also: Lexer::consume`

File: libnixf/lib/Parse/RangeBuilder.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "nixf/Basic/Range.h"
+
+#include <cassert>
+#include <stack>
+
+namespace nixf {
+class RangeBuilder {
```
Comment (Origami404) [inline libnixf/lib/Parse/RangeBuilder.h:9]: What is this class used for?

**Review by Origami404 — APPROVED** [review]

File: libnixf/include/nixf/Basic/Range.h (hunk 1 above)
Comment (inclyc) [reply libnixf/include/nixf/Basic/Range.h:12]: > I believe that we should define the meanings of these fields more carefully in the comments here, such as:

This should only be created by Lexer, I think. Maybe we can just make the constructor private.

**Pull request merged**